expression/aggregation/base_func.go

```diff
@@ -188,7 +188,8 @@ func (a *baseFuncDesc) typeInfer4MaxMin(ctx sessionctx.Context) {
 		a.RetTp = a.Args[0].GetType().Clone()
 		a.RetTp.Flag &^= mysql.NotNullFlag
 	}
-	if a.RetTp.Tp == mysql.TypeEnum || a.RetTp.Tp == mysql.TypeSet {
+	// TODO: fix other aggFuncs for TypeEnum & TypeSet
```

Comment:
According to https://github.com/pingcap/tidb/pull/14035#issuecomment-566505617, we do not need this `TODO`.
And we can add a annotation to describe why we need this judgment.

Reply:
Function `max()` for `enum` in aggregation is incorrect either, as represented in #13961 .
So I think `max`, `min`, `first_value`, `last_value`, `nth_value` for `enum` and `set` need such a fix too.
Maybe I can try to do it using `go:generate`.